pkg/upgrade/upgrades/first_upgrade.go

```diff
@@ -56,6 +56,9 @@ func FirstUpgradeFromRelease(
 	var batch catalog.DescriptorIDSet
 	const batchSize = 1000
 	if err := all.ForEachDescriptor(func(desc catalog.Descriptor) error {
+		if desc.GetName() == "foo" {
```

Comment:
so it does seem correct that this line would be reached, but for any descriptor that is not corrupt, the check right after this one should cause the repair to short-circuit. i think that should be good enough -- even the `invalid_objects` and `repairable_catalog_corruptions` tables are populated by first scanning all descriptors, then filtering only for the ones that have issues.

Reply:
Ah - ok so I left this in here (probably should've commented about this earlier) because I only wanted to debug the not corrupt table "foo" & this was the way I saw to do this with skipping all of the system(?) descriptors. However, the check right after this one does not cause the repair to short-circuit :(

So my confusion is: is it my testing (in TestUpgradeNoCorruption) that is causing this issue? I shall make a loom tomorrow and send it to you to be more ~illustrative~ :^)

Reply:
pairing up would be great!